elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala

```diff
@@ -271,47 +273,52 @@ object Validator extends Serializable {
         if (!accessor.validator.isInstanceOf[IgnoreValidator[_]]) {
           val name = new JStringBuilder(path.length + accessor.label.length)
             .append(path).append(accessor.label).toString
-          val o = if (accessor.validator.isInstanceOf[FieldValidator[_]]) {
+          val c = config.fieldConfig(name)
+          val (o, validationType) = if (accessor.validator.isInstanceOf[FieldValidator[_]]) {
             val v = accessor.validator.validateRecord(Unvalidated(accessor.value))
             val validationType = accessor.validator.asInstanceOf[FieldValidator[_]].validationType
+            (v, Some(validationType))
+          } else {
 
-            val c = config.fieldConfig(name)
-            if (v.isValid) {
-              if (c != NoCounter) {
-                reporter.reportValid(
-                  outermostClassName,
-                  name,
-                  validationType
-                )
-              }
-            } else if (v.isInvalid) {
-              if (c == ThrowException) {
-                throw new DataInvalidException(
-                  s"Invalid value ${v.forceGet.toString} found for field $path${accessor.label}")
+            val validationType =
+              if (accessor.validator.isInstanceOf[SeqLikeValidator[_, Seq]]) {
+                Some(accessor.validator.asInstanceOf[SeqLikeValidator[_, Seq]].validationType)
               }
-              if (c != NoCounter) {
-                reporter.reportInvalid(
-                  outermostClassName,
-                  name,
-                  validationType
-                )
+              else {
+                None
               }
-            }
-            v
-          } else {
-            accessor.validator.validateRecord(
+            val v = accessor.validator.validateRecord(
               Unvalidated(accessor.value),
               new JStringBuilder(path.length + accessor.label.length + 1)
                 .append(path).append(accessor.label).append(".").toString,
               Some(outermostClassName),
               config
             )
+            (v, validationType)
           }
 
           if (o.isValid) {
+            if (c != NoCounter && validationType.nonEmpty) {
```

Comment:
@idreeskhan pushed this down here to set the counters once if its a field validator or a SeqLike of field validators. I don't like the whole `(v, validationType)` above and checking here if validationType.nonEmpty but as of right now I don't have better code to only report counters if validation type is set. This way we can set counters for FieldValidator s and SeqLikeValidator only.

Reply:
If I understand correctly this will fail for `IgnoreValidators` and Records? I think it putting it here looks fine at a glance but can we add a comment somewhere about why it's here